From afc8c4948fcaaaea65e3e5291d3b06843f7ba1f0 Mon Sep 17 00:00:00 2001
From: Rafael Martins <rmartins@anaconda.com>
Date: Tue, 16 Jul 2024 23:11:22 +0200
Subject: [PATCH 4/6] chromium: force settings to build without xcode.

these changes are needed to allow chromium to build without using xcode. the patch
tries to use information from environment variables as much as possible, but version
information is hardcoded. this patch must be upgraded when trying to build with a
different macos sdk version.
---
 .../chromium/build/config/mac/mac_sdk.gni     | 23 ++++++++++---------
 .../build/toolchain/apple/toolchain.gni       |  4 ++--
 2 files changed, 14 insertions(+), 13 deletions(-)

diff --git a/src/3rdparty/chromium/build/config/mac/mac_sdk.gni b/src/3rdparty/chromium/build/config/mac/mac_sdk.gni
index dbf435f5d..422c767f6 100644
--- a/src/3rdparty/chromium/build/config/mac/mac_sdk.gni
+++ b/src/3rdparty/chromium/build/config/mac/mac_sdk.gni
@@ -104,10 +104,11 @@ if (use_system_xcode && (use_goma || use_remoteexec)) {
 }
 sdk_info_args += [ mac_sdk_name ]
 
-_mac_sdk_result = exec_script(script_name, sdk_info_args, "scope")
-xcode_version = _mac_sdk_result.xcode_version
-xcode_version_int = _mac_sdk_result.xcode_version_int
-xcode_build = _mac_sdk_result.xcode_build
+#_mac_sdk_result = exec_script(script_name, sdk_info_args, "scope")
+xcode_version = "1251"
+xcode_version_int = 1251
+xcode_build = "12E507"
+mac_sdk_path = getenv("CONDA_BUILD_SYSROOT")
 if (mac_sdk_path == "" && use_system_xcode && (use_goma || use_remoteexec)) {
   mac_sdk_path = _mac_sdk_result.sdk_path
 }
@@ -121,15 +122,15 @@ if (use_system_xcode) {
     "--print_sdk_build",
     mac_sdk_min,
   ]
-  find_sdk_lines =
-      exec_script("//build/mac/find_sdk.py", find_sdk_args, "list lines")
-  mac_sdk_version = find_sdk_lines[3]
-  mac_sdk_build_version = find_sdk_lines[2]
+#  find_sdk_lines =
+#      exec_script("//build/mac/find_sdk.py", find_sdk_args, "list lines")
+  mac_sdk_version = "11.3"
+  mac_sdk_build_version = "20E214"
   if (mac_sdk_path == "") {
-    mac_sdk_path = find_sdk_lines[0]
-    mac_bin_path = find_sdk_lines[1]
+#    mac_sdk_path = find_sdk_lines[0]
+#    mac_bin_path = find_sdk_lines[1]
   } else {
-    mac_bin_path = find_sdk_lines[1]
+    mac_bin_path = getenv("BUILD_PREFIX") + "/bin/" + getenv("HOST") + "-"
   }
 } else {
   mac_sdk_version = mac_sdk_official_version
diff --git a/src/3rdparty/chromium/build/toolchain/apple/toolchain.gni b/src/3rdparty/chromium/build/toolchain/apple/toolchain.gni
index c3560cc86..5dbdae1d2 100644
--- a/src/3rdparty/chromium/build/toolchain/apple/toolchain.gni
+++ b/src/3rdparty/chromium/build/toolchain/apple/toolchain.gni
@@ -157,8 +157,8 @@ template("single_apple_toolchain") {
     }
 
     prefix = rebase_path("$clang_base_path/bin/", root_build_dir)
-    _cc = "${prefix}clang"
-    _cxx = "${prefix}clang++"
+    _cc = getenv("BUILD_PREFIX") + "/bin/" + getenv("HOST") + "-clang"
+    _cxx = getenv("BUILD_PREFIX") + "/bin/" + getenv("HOST") + "-clang++"
 
     swiftmodule_switch = "-Wl,-add_ast_path,"
 
-- 
2.39.2 (Apple Git-143)

